share/spice/arxiv/arxiv.js

```diff
@@ -0,0 +1,71 @@
+(function (env) {
+    "use strict";
+    env.ddg_spice_arxiv = function(api_result){
+        var url = get_arxiv_rel_link( api_result.feed.entry.link );
+        var pdf_url = get_pdf_link( api_result.feed.entry.link );
+
+        if (!api_result) {
+            return Spice.failed('arxiv');
+        }
+
+        Spice.add({
+            id: "arxiv",
+
+            name: "Reference",
+            data: api_result,
+            meta: {
+                sourceName: "arxiv.org",
+                sourceUrl: url
+            },
+            normalize: function(item) {
+                return {
+                    title: sprintf( "%s (%s)",
+                        item.feed.entry.title.text,
+                        item.feed.entry.published.text.replace( /^(\d{4}).*/, "$1" )
+                    ),
+                    url: url,
+                    subtitle: item.feed.entry.author.map( function(e) {
+                        return e.name.text;
+                    } ).join(', '),
+                    description: item.feed.entry.summary.text
+                };
+            },
+            templates: {
+                group: 'info',
+                options: get_options( pdf_url )
+            }
+        });
+
+        function get_arxiv_rel_link(links) {
```

Comment:
@jbarrett we should move these out of callback, but keep them inside the closure.

Then they'll be defined outside out callback, but not added to the global namespace, and yet still accessible to our callback function :+1: 

```js
function (env) {
    "use strict";
    
    //private stuff here
    
    env.ddg_spice_arxiv = function (api_result) { ... }
}(this));
```